src/main/java/no/fintlabs/resourceserver/security/SecurityConfiguration.java

```diff
@@ -0,0 +1,185 @@
+//                        .jwtAuthenticationConverter(new FintJwtUserConverter())
+//                )
+//                .authorizeExchange()
+//                .anyExchange()
+//                .access((Mono<Authentication> authenticationMono, AuthorizationContext object) -> authenticationMono.map(authentication -> {
+//                            Set<String> authorizedOrgIdAuthorities = adminApiSecurityProperties.getAuthorizedOrgIds()
+//                                    .stream()
+//                                    .map(orgId -> "ORGID_" + orgId)
+//                                    .collect(Collectors.toSet());
+//
+//                            Set<String> grantedAuthorities = authentication
+//                                    .getAuthorities()
+//                                    .stream()
+//                                    .map(GrantedAuthority::getAuthority)
+//                                    .collect(Collectors.toSet());
+//
+//                            return !Collections.disjoint(grantedAuthorities, authorizedOrgIdAuthorities)
+//                                    && !(adminApiSecurityProperties.isRequireAdminRole() && !grantedAuthorities.contains("ROLE_admin"));
+//                        }
+//                ).map(AuthorizationDecision::new))
+//                .and()
+//                .build();
+//    }
+
+    @Order(3)
+    @Bean
+    SecurityWebFilterChain externalApiFilterChain(
+            ExternalApiSecurityProperties externalApiSecurityProperties,
+            ServerHttpSecurity http
+    ) {
+        http
+                .securityMatcher(new PathPatternParserServerWebExchangeMatcher(UrlPaths.EXTERNAL_API + "/**"))
+                .addFilterBefore(new AuthorizationLogFilter(), SecurityWebFiltersOrder.AUTHENTICATION);
+
+        if (!externalApiSecurityProperties.isEnabled()) {
+            return denyAll(http);
+        }
+
+        return externalApiSecurityProperties.isPermitAll()
+                ? permitAll(http)
+                : http
+                .oauth2ResourceServer((resourceServer) -> resourceServer
+                        .jwt()
+                        .jwtAuthenticationConverter(clientJwtConverter)
+                )
+                .authorizeExchange()
+                .anyExchange()
+                .hasAnyAuthority(mapToAuthoritiesArray(
+                        SOURCE_APPLICATION_ID_PREFIX,
+                        externalApiSecurityProperties.getAuthorizedClientIds()
+                ))
+                .and()
+                .build();
+    }
+
+    @Order(4)
+    @Bean
+    SecurityWebFilterChain globalFilterChain(
+            ServerHttpSecurity http
+    ) {
+        http.addFilterBefore(new AuthorizationLogFilter(), SecurityWebFiltersOrder.AUTHENTICATION);
+        return denyAll(http);
+    }
+
+    private SecurityWebFilterChain permitAll(ServerHttpSecurity http) {
+        return http
+//                .csrf().disable() // TODO: 22/08/2022 Enable/disable?
```

Comment:
Denne er nødvendig å ha med for å åpne for tilgang i pwf demo. Det er ikke så bra når det er endepunkter som bruker av nettlesere. Se om vi kan fikse slik at vi ikke trenger å disable den for åpen tilgang i demo?